src/Table.cpp

```diff
@@ -120,32 +132,59 @@ TableValue::ValueType TableValue::GetType() const {
   return static_cast<ValueType>(m_impl->m_value.which());
 }
 
-bool TableValue::GetBool() const { return boost::get<bool>(m_impl->m_value); }
+bool TableValue::GetBool() const {
+  return boost::get<bool>(m_impl->m_value);
+}
+
+
+boost::uint8_t TableValue::GetUint8() const {
+  return boost::get<boost::uint8_t>(m_impl->m_value);
+}
 
 boost::int8_t TableValue::GetInt8() const {
   return boost::get<boost::int8_t>(m_impl->m_value);
 }
 
+boost::uint16_t TableValue::GetUint16() const {
+  return boost::get<boost::uint16_t>(m_impl->m_value);
+}
+
 boost::int16_t TableValue::GetInt16() const {
   return boost::get<boost::int16_t>(m_impl->m_value);
 }
 
+boost::uint32_t TableValue::GetUint32() const {
+  return boost::get<boost::uint32_t>(m_impl->m_value);
+}
+
 boost::int32_t TableValue::GetInt32() const {
   return boost::get<boost::int32_t>(m_impl->m_value);
 }
 
+boost::uint64_t TableValue::GetUint64() const {
+  return boost::get<boost::uint64_t>(m_impl->m_value);
+}
+
 boost::int64_t TableValue::GetInt64() const {
   return boost::get<boost::int64_t>(m_impl->m_value);
 }
 
 boost::int64_t TableValue::GetInteger() const {
   switch (m_impl->m_value.which()) {
+    case VT_uint8:
+      return GetUint8();
     case VT_int8:
       return GetInt8();
+    case VT_uint16:
+      return GetUint16();
     case VT_int16:
       return GetInt16();
+    case VT_uint32:
+      return GetUint32();
     case VT_int32:
       return GetInt32();
+    case VT_uint64:
+      return GetUint64();
```

Comment:
A third option would be to throw a `std::overflow_error` if the result of `GetUint64()` is out of range.

I agree that simply leaving off uint64_t (option 1) makes for a non-obvious API.  Between options 2 and 3, I guess it depends on how we expect users to use this API.  

Fewer functions make thinking about signed vs unsigned easier, and if users just want a 'integer' then its easier to use. If they expect large integers then, they should probably be using `GetUint64()` directly.

A `GetUnsignedInteger()` makes one less runtime exception to have to deal with and potentially handle if you're looking to handle every case.

I'm leaning towards option 3 given this is intended to be convenience method, however if your use-case demands that it work across the entire range of `uint64_t` values, I'll accept option 2.